api/utils.go

```diff
@@ -428,3 +428,48 @@ func normalizeEvent(receivedEvent *cluster.Event) ([]byte, error) {
 
 	return data, nil
 }
+
+// GetHTTPErrorStatusCode takes an error, checks a list of known errors,
+// and return the HTTP status code.
+// Returns 500 if there is no match
+// List of errors are extracted from https://github.com/moby/moby/blob/17.05.x/api/server/httputils/errors.go#L56-L68
+// In engine > 18.x, we moved to typed errors check, which is hard to extract the err strings for
+func GetHTTPErrorStatusCode(err error) int {
```

Comment:
Wonder if, instead of copying this, we could use the upstream package (possibly would need some small refactoring to prevent pulling in too much code?)

Reply:
@thaJeztah 
I did look into this approach, but the scope was larger for what we are trying to accomplish for this one request.
I am happy to start using errdefs if there is more need to set the http status code for other errors